monai/bundle/config_item.py

```diff
@@ -262,34 +282,78 @@ def resolve_args(self):
 
         """
-        _is_disabled = self.get_config().get("_disabled_", False)
+        _is_disabled = self.get_config().get(ConfigComponentReservedKeys.DISABLED, False)
         return _is_disabled.lower().strip() == "true" if isinstance(_is_disabled, str) else bool(_is_disabled)
 
+    def _get_wrapper(self) -> None | Callable[[object], object]:
+        """
+        Utility function used in `instantiate()` to check if a wrapper is specified in the config.
+
+        """
+        wrapper = self.get_config().get(ConfigComponentReservedKeys.WRAPPER, None)
+        if _wrapper_feature_flag.enabled:
+            if wrapper is not None:
+                if callable(wrapper):
+                    return wrapper  # type: ignore
+                else:
+                    raise ValueError(
+                        f"wrapper must be a callable, but got type {type(wrapper)}: {wrapper}."
+                        "make sure all references are resolved before calling instantiate "
+                        "and the wrapper is a callable."
+                    )
+        elif wrapper is not None:
+            warnings.warn(
+                f"ConfigComponent: {self.get_id()} has a key {ConfigComponentReservedKeys.WRAPPER}. "
+                "Since the feature flag CONFIG_WRAPPER is not enabled, the key will be treated as a normal config key. "
+                "In future versions of MONAI, this key might be reserved for the wrapper functionality."
+            )
+        return None
+
     def instantiate(self, **kwargs: Any) -> object:
         """
         Instantiate component based on ``self.config`` content.
         The target component must be a `class` or a `function`, otherwise, return `None`.
 
         Args:
-            kwargs: args to override / add the config args when instantiation.
+            kwargs: instantiate_kwargs to override / add the config instantiate_kwargs when instantiation.
 
         """
         if not self.is_instantiable(self.get_config()) or self.is_disabled():
             # if not a class or function or marked as `disabled`, skip parsing and return `None`
             return None
 
         modname = self.resolve_module_name()
-        mode = self.get_config().get("_mode_", CompInitMode.DEFAULT)
-        args = self.resolve_args()
-        args.update(kwargs)
-        return instantiate(modname, mode, **args)
+        mode = self.get_config().get(ConfigComponentReservedKeys.MODE, CompInitMode.DEFAULT)
+        instantiate_kwargs = self.resolve_args()
+        instantiate_kwargs.update(kwargs)
+        wrapper = self._get_wrapper()
+        if wrapper is not None:
+            return wrapper(instantiate(modname, mode, **instantiate_kwargs))
+
+        try:
+            return instantiate(modname, mode, **instantiate_kwargs)
+        except Exception as e:
+            if _wrapper_feature_flag.enabled and self.get_id().endswith(ConfigComponentReservedKeys.WRAPPER):
```

Comment:
`self.get_id()` will only return a str, will this check worked?

Reply:
```
>>> "test::_wrapper_".endswith(ConfigComponentReservedKeys.WRAPPER)
True
```
Yes this should work, this evaluates to true for me